lib/analyzer/image-inspector.ts

```diff
@@ -152,17 +155,116 @@ async function pullImage(
       platform,
     );
     if (pullAndSaveSuccessful) {
-      return;
+      imageNames.push(targetImage);
+      return imageNames;
     }
   }
 
-  await pullFromContainerRegistry(
+  const { indexDigest, manifestDigest } = await pullFromContainerRegistry(
     docker,
     targetImage,
     imageSavePath,
     username,
     password,
   );
+
+  const imageNameWithIndexDigest = getImageNameWithIndexDigest(
+    indexDigest,
+    targetImage,
+  );
+  const imageNameWithManifestDigest = getImageNameWithManifestDigest(
+    manifestDigest,
+    targetImage,
+  );
+
+  return getImageNames(
+    imageNameWithIndexDigest,
+    imageNameWithManifestDigest,
+    targetImage,
+  );
+}
+
+export function getImageNames(
+  indexImageName: string | undefined,
+  manifestImageName: string | undefined,
+  targetImage: string,
+): string[] {
+  const imageNames: Set<string> = new Set();
+
+  if (indexImageName) {
+    imageNames.add(indexImageName);
+  }
+  if (manifestImageName) {
+    imageNames.add(manifestImageName);
+  }
+  if (indexImageName !== targetImage && manifestImageName !== targetImage) {
+    imageNames.add(targetImage);
+  }
+  return Array.from(imageNames);
+}
+
+export function getImageNameWithIndexDigest(
```

Comment:
If you `s/index/manifest/g`, you get the exact same function as `getImageNameWithManifestDigest` 😉 
Or am I missing something? 😅 

I don't think we need two different functions for that 😄 Simply call it `getImageNameWithDigest`, or even just `getImageName`?

Reply:
(If this is all extracted and put into a class, it doesn't really matter / would need a different API anyway.)